packages/core/src/fhir-deduplication/deduplicate-fhir.ts

```diff
@@ -519,3 +465,175 @@ function removeDuplicateReferences<T extends Resource>(entry: T): T {
 
   return entry;
 }
+
+/**
+ * Finds and updates references to the deduplicated resources
+ */
+function replaceResourceReferences(
+  resourceArrays: ExtractedFhirTypes,
+  refReplacementMap: Map<string, string>,
+  resourceType?: string[]
+): ExtractedFhirTypes {
+  return Object.fromEntries(
+    Object.entries(resourceArrays).map(([key, resources]) => {
+      if (resourceType && !resourceType.includes(key)) {
+        return [key, resources];
+      }
+      if (Array.isArray(resources)) {
+        return [
+          key,
+          resources.map(resource => replaceResourceReference(resource, refReplacementMap)),
+        ];
+      } else if (resources) {
+        // Handle single resource case
+        return [key, replaceResourceReference(resources, refReplacementMap)];
+      } else {
+        // Handle null or undefined case
+        return [key, resources];
+      }
```

Comment:
it can be simplifid more by just ignoring the case where its not array. the only resource that applies to is patient and there is nothing to replace or deduplicate in patient